packages/mobx-react-lite/src/useAsObservableSource.ts

```diff
@@ -7,7 +7,7 @@ export function useAsObservableSource<TSource extends object>(current: TSource):
         useDeprecated(
             "[mobx-react-lite] 'useAsObservableSource' is deprecated, please store the values directly in an observable, for example by using 'useLocalObservable', and sync future updates using 'useEffect' when needed. See the README for examples."
         )
-    const [res] = useState(() => observable(current, {}, { deep: false }))
+    const res = useState(() => observable(current, {}, { deep: false }))[0]
```

Comment:
NIT but also kinda important in this case, please add a comment here why idiomatic destructuring isn't used here.

Reply:
Important indeed. I added a comment to explain the reasoning.